### ISSUE TITLE
what is the difference between this and awx.awx and ansible.tower collections?

### ISSUE THREAD

**Issue by itewk** [redhat-cop/automate_tower_genie_collections#165]
What is the difference and/or value add of this collection over the following other existing collections?

* [ansibe.tower](https://cloud.redhat.com/ansible/automation-hub/repo/published/ansible/tower) collection
* [awx.awx](https://galaxy.ansible.com/awx/awx) collection

**Comment by sean-m-sullivan** [comment]
The short answer is this is a solution for Infrastructure as Code, or what we have been calling Tower as code. 

It allows for a data model that can be consumed to describe every object api endpoint in Tower. Where as the referenced collections just allow for interaction with those objects. 

This utilizes Either the awx.awx collection or ansible.tower collection and allows for a data model to be followed to update them. its a wrapper on those modules for easy consumption. For use we generally use something [similar to files in the tower configs playbook directory. ](https://github.com/redhat-cop/tower_configuration/tree/devel/playbooks/tower_configs) and invoke them similar to the [playbook](https://github.com/redhat-cop/tower_configuration/blob/devel/playbooks/configure_tower.yml), or just add roles depending on what you need to do. 

For example We have custom surveys that we populate via an API that we have templated, so we run that and get variables set for the templates before invoking the job template and workflow roles. 

What had happened was that some consultants are Redhat had found that we had implemented the same general process for consuming/using those modules, but had 3 different ways of doing it. So we put the best ideas out of all them, and combined them here, refining and updating as we get input and things change. The goal is an out of the box, canned, solution for people to use for Tower as Code. 

Doing this helps prevent drift in tower and prevent manual changes. Generally we run this at least once a day if not more on my companies tower, Changes can now be tracked in Github as well. 

Another good example of this is what has been done at [ah_configuration](https://github.com/redhat-cop/ah_configuration), which under the hood uses the concepts and a lot of the code from both awx.awx and this and combines them into one collection, but for automation hub.

**Comment by itewk** [comment]
@sean-m-sullivan is there a way to swap whether using awx.awx or ansible.twoer? and is there technical reasons to do so, or just support reasons? is ansible.tower just a down stream / rebrand of awx.awx?

**Comment by sean-m-sullivan** [comment]
To swap you would just change the collection referenced in the top of the playbook
```
---
- name: Playbook to configure ansible tower post installation
  hosts: localhost
  connection: local
  vars:
    tower_validate_certs: false
  collections:
    - awx.awx # or ansible.tower
 ```
 we made a change a while back to allow people to choose which one they wanted to use. 
 
 In terms of technical reasons, there really is not one. They are compiled from the same source, but only one is "supported" as its on the redhat automation hub. But from what I've seen if the latest version of awx.awx was released Tuesday, and the latest tower was released a week later, The tower would have the latest updates, and vice versa depending on when each was released. Barring any changes changes not yet it tower api.
